src/lib/graphql_lib/tests.ml

```diff
@@ -0,0 +1,235 @@
+  | `Assoc [ ("data", `Assoc [ ("test", value) ]) ] ->
+      value
+  | json ->
+      failwithf "(%s) Unexpected format of JSON response:%s" __LOC__
+        (Yojson.Basic.to_string json)
+        ()
+
+module Make_test (S : Scalars.Json_intf) (G : Test_Intf with type t = S.t) =
+struct
+  let query_server_and_compare value =
+    let schema =
+      Graphql_async.Schema.(
+        schema
+          [ field "test"
+              ~typ:(non_null @@ S.typ ())
+              ~args:Arg.[]
+              ~resolve:(fun _ () -> value)
+          ])
+    in
+    test_query schema () "{ test }" (fun response ->
+        [%test_eq: G.t] value (S.parse @@ get_test_field response) )
+
+  let%test_unit "test" =
+    Quickcheck.test G.gen ~sexp_of:G.sexp_of_t ~f:query_server_and_compare
+end
+
+module UInt32_gen = struct
+  include Unsigned.UInt32
+
+  let gen =
+    Int32.quickcheck_generator
+    |> Quickcheck.Generator.map ~f:Unsigned.UInt32.of_int32
+
+  let sexp_of_t = Fn.compose Int32.sexp_of_t Unsigned.UInt32.to_int32
+end
+
+module UInt64_gen = struct
+  include Unsigned.UInt64
+
+  let gen =
+    Int64.quickcheck_generator
+    |> Quickcheck.Generator.map ~f:Unsigned.UInt64.of_int64
+
+  let sexp_of_t = Fn.compose Int64.sexp_of_t Unsigned.UInt64.to_int64
+end
+
+module String_gen = struct
+  include String
+
+  let gen = gen_nonempty
+end
+
+module Span_gen = struct
+  include Time.Span
+
+  let gen =
+    let open Core_kernel_private.Span_float in
+    let millenium = of_day (Float.round_up (365.2425 *. 1000.)) in
+    Quickcheck.Generator.filter quickcheck_generator ~f:(fun t ->
+        neg millenium <= t && t <= millenium )
+
+  let compare x y =
+    (* https://github.com/janestreet/core_kernel/blob/v0.14.1/src/float.ml#L61 *)
+    (* Note: We have to use a different tolerance than
+       `Core_kernel.Time.Span.robustly_compare` does
+       because spans are rounded to the millisecond when
+       serialized through GraphQL. See the implementation
+       of Span in the `Graphql_basic_scalars` module. *)
+    let tolerance = 1E-3 in
+    let diff = x - y in
+    if diff < of_sec ~-.tolerance then -1
+    else if diff > of_sec tolerance then 1
+    else 0
+end
+
+module Time_gen = struct
+  type t = Core_kernel.Time.t
+
+  (* The following generator function is copied from version 0.15.0 of the core library, and only generates values that can be serialized.
+     https://github.com/janestreet/core/blob/35941320a3eab628786ae3853e5f753a3ab357c2/core/src/span_float.ml#L742-L754.
+     See issue https://github.com/MinaProtocol/mina/issues/11310.
+     Once the core library is updated to >= 0.15.0, [Core.Time.quickcheck_generator] should be used instead work.*)
+
+  let gen =
+    Quickcheck.Generator.map Span_gen.gen
+      ~f:Core_kernel.Time.of_span_since_epoch
+
+  let sexp_of_t = Core.Time.sexp_of_t
+
+  let compare x y = Core_kernel.Time.robustly_compare x y
+end
+
+module Slot_gen = struct
+  include Consensus__Slot
```

Comment:
```suggestion
  include Consensus.Slot
```

Reply:
I think these double underscore modules are not meant to be used directly.
(There are a few others as well).